### PR TITLE
Cleanup and refactor

### PR DESCRIPTION
**Pull request by sidoshi**

### CODE REVIEW

File: src/api/users/utils.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as t from 'io-ts'
```
Comment (L-oris) [inline src/api/users/utils.ts]: silly question: based on what we discussed yesterday, should this file be a service?

File: src/framework.ts [hunk 2]
```diff
@@ -65,31 +62,27 @@ const toKoaHandler = (config: RouteConfig) => {
...
-const addRouteToRouter = (router: Router, routeConfig: RouteConfig): Router => {
-    return router[getRouterMethod(routeConfig)](routeConfig.path, toKoaHandler(routeConfig))
+function registerRouteFromConfig(koaRouter: Router, routeConfig: RouteConfig) {
+    koaRouter[getRouterMethod(routeConfig)](routeConfig.path, toKoaHandler(routeConfig))
 }
 
-const addRoutesToRouter = (router: Router, routesConfig: RouteConfig[]): Router =>
-    routesConfig.reduce<Router>(addRouteToRouter, router)
+export function createRouter(opt?: Router.IRouterOptions) {
+    const koaRouter = createKoaRouter(opt)
```
[the remark this answers is not in the document]
Comment (L-oris) [reply src/framework.ts]: PS. I've written this comment before creating the PR #10 , it's outdated now

File: src/api/subreddits/create.route.ts [hunk 3]
```diff
@@ -17,7 +18,7 @@ const createSubredditRequestBodyValidator = t.intersection([
     }),
 ])
 
-const createSubredditResponseBodyValidator = t.intersection([
+const responseBodySchema = t.intersection([
```
[the remark this answers is not in the document]
Comment (sidoshi) [reply src/api/subreddits/create.route.ts]: Maybe there is some confusion, I thought it was discussed that each file will have only one endpoint.

File: src/api/users/utils.ts (hunk 1 above)
Comment (sidoshi) [reply src/api/users/utils.ts]: Yes, this should be a service. will update

File: src/api/subreddits/create.route.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (L-oris) [reply src/api/subreddits/create.route.ts]: Oh great! I was pushing to that direction, too.
Forget this comment

**Pull request merged**